src/server/tiered_storage.cc

```diff
@@ -124,49 +112,34 @@ class TieredStorage::ShardOpManager : public tiering::OpManager {
     }
   }
 
-  bool ReportFetched(EntryId id, string_view value, tiering::DiskSegment segment,
+  bool NotifyFetched(EntryId id, string_view value, tiering::DiskSegment segment,
                      bool modified) override;
 
-  bool ReportDelete(tiering::DiskSegment segment) override {
-    if (OccupiesWholePages(segment.length))
-      return true;
-
-    auto bin = ts_->bins_->Delete(segment);
-    if (bin.empty) {
-      return true;
-    }
-
-    if (bin.fragmented) {
-      // Trigger read to signal need for defragmentation. ReportFetched will handle it.
-      VLOG(1) << "Enqueueing bin defragmentation for: x" << bin.segment.offset;
-      Enqueue(kFragmentedBin, bin.segment, [](std::string*) { return false; });
-    }
+  bool NotifyDelete(tiering::DiskSegment segment) override;
 
-    return false;
+  // Set value to be an in-memory type again, either empty or with a value. Update memory stats
+  void Upload(DbIndex dbid, string_view value, size_t serialized_len, PrimeValue* pv) {
```

Comment:
Renamed from SetInMemory. 
Removed the logic that handles deletes:
```
    pv->Reset();
    if (!value.empty())
      pv->SetString(value);   
```

Reply:
The following PR will introduce storing/fetching encoded blobs without decoding, as you suggested